### PR TITLE
Return domains for _find_domains_or_certname

### LINKED ISSUE #3935: --cert-name causes explosions when trying to use "run" as an installer
One thing you can do with the `run` verb is try to install an existing cert (since the `install` verb is more like a construction kit than a real Certbot subcommand). 
```
certbot certificates
Saving debug log to /var/log/letsencrypt/letsencrypt.log

-------------------------------------------------------------------------------
Found the following certs:
  Certificate Name: isnot.org
    Domains: isnot.org goxogle.com is.isnot.org www.isnot.org
    Expiry Date: 2017-03-20 17:57:00+00:00 (VALID: 89 days)
    Certificate Path: /etc/letsencrypt/live/isnot.org/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/isnot.org/privkey.pem
...

certbot --cert-name isnot.org --debug
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Starting new HTTPS connection (1): acme-v01.api.letsencrypt.org
Cert not yet due for renewal

You have an existing certificate that has exactly the same domains or certificate name you requested and isn't close to expiry.
(ref: /etc/letsencrypt/renewal/isnot.org.conf)

What would you like to do?
-------------------------------------------------------------------------------
1: Attempt to reinstall this existing certificate
2: Renew & replace the cert (limit ~5 per 7 days)
-------------------------------------------------------------------------------
Select the appropriate number [1-2] then [enter] (press 'c' to cancel): 1
Keeping the existing certificate
Traceback (most recent call last):
  File "venv/bin/certbot", line 11, in <module>
    load_entry_point('certbot', 'console_scripts', 'certbot')()
  File "/home/ubuntu/letsencrypt/certbot/main.py", line 831, in main
    return config.func(config, plugins)
  File "/home/ubuntu/letsencrypt/certbot/main.py", line 570, in run
    lineage.chain, lineage.fullchain)
  File "/home/ubuntu/letsencrypt/certbot/client.py", line 380, in deploy_certificate
    for dom in domains:
TypeError: 'NoneType' object is not iterable
```

### PR DESCRIPTION
**Pull request by ohemorange**
Fixes #3935.

### CODE REVIEW

File: certbot/main.py [hunk 1]
```diff
@@ -294,6 +294,9 @@ def _find_domains_or_certname(config, installer):
                            "will help in domain names autodiscovery, or "
                            "--cert-name for an existing certificate name.")
 
+    if config.certname and not domains:
```
Comment (pde) [inline certbot/main.py]: I think this can be a third stanza in the `if .. elif` above?

**Comment by ohemorange** [comment]
Any idea why coveralls is lying?

Cover says:
`certbot/main.py                                403     44    89%   129-157, 178-205, 225, 231, 235-236, 264, 272, 373, 387-389, 524, 555-556, 568, 637-645, 739, 844-846`

I modified 287-306. But coveralls is saying missed +1. Is my brain's range intersection module not working today or...?

File: certbot/main.py (hunk 1 above)
Comment (ohemorange) [inline certbot/main.py]: Revamped this whole section.

**Comment by pde** [comment]
Possibly the issue Brad fixed in #3919, which I just merged?  Let me see what happens when I kick Travis.

**Comment by ohemorange** [comment]
No idea what's going on with coveralls, but manually checking cover looks fine. Merging!

**Pull request merged**